### PR TITLE
Support for syslognet: logging to a remote syslog server through TCP

### PR DESCRIPTION
**Pull request by philaris**
Thank you for the nice package! I have implemented appender and layout functions (`appender_syslognet`, `layout_syslognet`) that can be used to interface with the recently uploaded to CRAN `syslognet` R package that allows logging by sending TCP messages to a remote server running syslog.

### CODE REVIEW

File: R/appenders.R [hunk 1]
```diff
@@ -426,4 +426,31 @@ appender_async <- function(appender, batch = 1, namespace = 'async_logger',
...
+#' @export
+#' @note This functionality depends on the \pkg{syslognet} package.
+#' @examples \dontrun{
+#' if (requireNamespace("syslognet", quietly = TRUE)) {
+#'   log_appender(appender_syslognet("test_app", 'remoteserver'))
+#'   log_info("Test message.")
+#' }
+#' }
+appender_syslognet <- function(identifier, server, port = 601L) {
+  logger::fail_on_missing_package('syslognet')
```
Comment (daroczig) [inline R/appenders.R:444]: Sorry, just saw your PR -- first of all, thanks a lot for your contribution!

Not sure if the double colon is needed here:

```suggestion
  fail_on_missing_package('syslognet')
```

On the other hand, we need to `force` the `identifier`, `server` and `port` params

File: R/appenders.R [hunk 2]
```diff
@@ -426,4 +426,31 @@ appender_async <- function(appender, batch = 1, namespace = 'async_logger',
...
+#' @return A function taking a \code{lines} argument.
+#' @export
+#' @note This functionality depends on the \pkg{syslognet} package.
+#' @examples \dontrun{
+#' if (requireNamespace("syslognet", quietly = TRUE)) {
+#'   log_appender(appender_syslognet("test_app", 'remoteserver'))
+#'   log_info("Test message.")
+#' }
+#' }
+appender_syslognet <- function(identifier, server, port = 601L) {
```
Comment (daroczig) [inline R/appenders.R:443]: can you please move this function up to right below `appender_syslog`?

File: DESCRIPTION [hunk 3]
```diff
@@ -6,7 +6,7 @@ Authors@R: c(
     )
 Title: A Lightweight, Modern and Flexible Logging Utility
 Description: Inspired by the the 'futile.logger' R package and 'logging' Python module, this utility provides a flexible and extensible way of formatting and delivering log messages with low overhead.
-Version: 0.1.9000
```
Comment (daroczig) [inline DESCRIPTION:9]: pls let's not bump the version as part of this PR

**Comment by philaris** [comment]
Hi @daroczig, I agree and I have made the fixes that you suggested.

File: R/appenders.R [hunk 4]
```diff
@@ -255,6 +255,34 @@ appender_syslog <- function(identifier, ...) {
...
+  fail_on_missing_package('syslognet')
```
Comment (daroczig) [inline R/appenders.R:273]: we still need to force the `appender_syslognet` params to be passed down to the closure:

```suggestion
  fail_on_missing_package('syslognet')
  force(server)
  force(port)
```

File: R/layouts.R [hunk 5]
```diff
@@ -250,3 +250,35 @@ layout_json_parser <- function(fields = c('time', 'level', 'ns', 'ans', 'topenv'
...
+#' Format a log record for syslognet.
+#' This function converts the logger log level to a
+#' log severity level according to RFC 5424 "The Syslog Protocol".
+#'
+#' @param level level of log message.
+#' @param msg string of log message.
+#' @param namespace namespace.
+#' @param .logcall logcall; see \code{\link[logger]{layout_blank}}.
+#' @param .topcall topcall; see \code{\link[logger]{layout_blank}}.
+#' @param .topenv topenv; ; see \code{\link[logger]{layout_blank}}.
```
Comment (daroczig) [inline R/layouts.R:276]: we can simplify this I think via:

```suggestion
#' @inheritParams layout_simple
```

File: R/layouts.R [hunk 6]
```diff
@@ -250,3 +250,35 @@ layout_json_parser <- function(fields = c('time', 'level', 'ns', 'ans', 'topenv'
...
+#' @return A character vector with a severity attribute.
+#' @export
+layout_syslognet <- function(level, msg, namespace = NA_character_,
```
Comment (daroczig) [inline R/layouts.R:279]: can you pls wrap this in a `structure` (like all the other layouts) to have a `generator`?

**Comment by daroczig** [comment]
Thanks a lot, @philaris :bow: 
Much appreciated!

Sorry, missed a few things -- could you please take care of those as well before merging?

**Comment by philaris** [comment]
@daroczig I implemented the rest of your suggestions. Please check.

File: R/layouts.R [hunk 7]
```diff
@@ -250,3 +250,33 @@ layout_json_parser <- function(fields = c('time', 'level', 'ns', 'ans', 'topenv'
...
+#' @return A character vector with a severity attribute.
+#' @export
+layout_syslognet <- structure(
+  function(level, msg, namespace = NA_character_,
+           .logcall = sys.call(), .topcall = sys.call(-1), .topenv = parent.frame()) {
+    ret <- paste(attr(level, "level"), msg)
+    attr(ret, 'severity') <- level_to_severity(level)
+    return(ret)
+  },
+  generator = quote(layout_simple())
```
Comment (daroczig) [inline R/layouts.R:281]:
```suggestion
  generator = deparse(match.call())
```

**Review by daroczig — APPROVED** [review]
thanks a lot, merging -- will need to update the `NEWS.md` etc but I can do that later mentioning you as well :bow:

**Comment by daroczig** [comment]
Hm, there's something failing on Travis, investigating and will be able to merge after that.

**Comment by philaris** [comment]
HI @daroczig, there was a problem with the layout_syslognet generator. I made its generator like the one used in layout_logging. Now there seem to be no errors.

**Comment by daroczig** [comment]
Awesome thanks -- sorry for the direct commits in your branch, but these small tweaks were required to make the code coverage tests happy .. seemed easier to commit directly than asking you to do so.

**Comment by philaris** [comment]
No problem, thanks @daroczig !

**Pull request merged**